pkg/config/config.go

```diff
@@ -390,6 +390,9 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("dogstatsd_entity_id_precedence", false)
 	// Sends Dogstatsd parse errors to the Debug level instead of the Error level
 	config.BindEnvAndSetDefault("dogstatsd_disable_verbose_logs", false)
+	// Location to store dogstatsd captures by default
+	defaultDogstatsdCapturePath := path.Join(config.GetString("run_path"), "dsd_capture")
```

Comment:
This will not actually pick up the configured `run_path` value because when this init function runs the config file and env vars have not be loaded yet.

At the moment we don't have a built-in way to make some config options' defaults depend on other config options' configured values. I think the simplest option here is to leave the default here to `""`, and have the logic in `pkg/dogstatsd/replay/capture.go` default to `path.Join(config.GetString("run_path"), "dsd_capture")` if the `dogstatsd_capture_path` config value is empty. It has the downside of "leaking" some config logic outside of the `config` package, but I think this is an acceptable downside.

Reply:
I tested this and it worked though. It's also the same approach used for the forwarder storage path, which also works.  🤔  But you're right in what you're saying about the `init()` logic.

Reply:
It works as long as the `run_path` is not configured to a custom value in the config file/env vars. If the `run_path` is configured to a custom value, then I think the expectation is that the default dogstatsd capture path would use that customized `run_path` value as the base dir, but in reality I think it'll still use the default run path as the base dir.

The forwarder storage path is semi-broken in the same way, I'll open a card in backlog so that we address this.

Reply:
Of course! 🤦